tests/bin/opam-file-locations/run.t

```diff
@@ -0,0 +1,100 @@
+Set up a project with an `.opam` file in the toplevel folder:
+
+  $ cat > CHANGES.md << EOF
+  > ## 0.1.0
+  > 
+  > - Initial release
+  > 
+  > EOF
+  $ cat > dune-project << EOF
+  > (lang dune 3.8)
+  > (name myproject)
+  > EOF
+  $ cat > myproject.opam << EOF
+  > opam-version: "2.0"
+  > EOF
+  $ git init 2> /dev/null . > /dev/null
+  $ touch README LICENSE
+  $ cat > .gitignore << EOF
+  > _build
+  > .bin
+  > /dune
+  > run.t
+  > EOF
+  $ git add CHANGES.md README LICENSE *.opam dune-project .gitignore
+  $ git commit -m 'Initial commit' > /dev/null
+
+Tagging should work
+
+  $ dune-release tag -y
+  [-] Extracting tag from first entry in CHANGES.md
+  [-] Using tag "0.1.0"
+  [+] Tagged HEAD with version 0.1.0
+
+`dune-release distrib` should work.
+
+  $ dune-release distrib --skip-lint | make_dune_release_deterministic
+  [-] Building source archive
+  [+] Wrote archive _build/myproject-0.1.0.tbz
+  
+  [-] Building package in _build/myproject-0.1.0
+  [ OK ] package(s) build
+  
+  [-] Running package tests in _build/myproject-0.1.0
+  [ OK ] package(s) pass the tests
+  
+  [+] Distribution for myproject 0.1.0
+  [+] Commit <deterministic>
+  [+] Archive _build/myproject-0.1.0.tbz
+
+Now let's move the `.opam` file to the `opam/` subfolder. OPAM supports `.opam`
+files in the `opam/` subfolder, but for dune to pick it up we need to tell it
+to look in that folder.
```

Comment:
That's very clear. Does dune 3.8 support generating files in `opam/` ? That could also be a nice e2e test.

Reply:
It does (and it works)! I was a bit on the fence, because it feels a bit like testing dune and not dune release, but I guess this won't hurt :)

This made me think of a future where the repos would not have the `opam` files pregenerated but rather dune-release would create them just for the tarball. Though that introduces some issues with pinning.